examples/active-class-name/components/ActiveLink.js

```diff
@@ -4,15 +4,15 @@ import Link from 'next/link'
 import React, { Children } from 'react'
 
 const ActiveLink = ({ children, activeClassName, ...props }) => {
-  const { asPath } = useRouter()
+  const { pathname } = useRouter()
   const child = Children.only(children)
   const childClassName = child.props.className || ''
 
   // pages/index.js will be matched via props.href
   // pages/about.js will be matched via props.href
   // pages/[slug].js will be matched via props.as
   const className =
-    asPath === props.href || asPath === props.as
+    pathname === props.href || pathname === props.as
       ? `${childClassName} ${activeClassName}`.trim()
       : childClassName
 
```

Comment:
This implementation doesn't recognize dynamic route. Here is [demo](https://stackblitz.com/edit/github-w7g78q?file=components/ActiveLink.js). If we try to change the url to `/something`, `[slug]` will no longer active

Reply:
Here is the [link](https://stackblitz.com/edit/github-mvrqrn?file=components/ActiveLink.js) of implementation using `pathname`.

Reply:
Both demos appear to be matching the dynamic route correctly, the problem with the `pathname` approach is that it will match multiple `ActiveLink`s for a dynamic route but different `as` values when it should only match one or the other e.g. 

`<ActiveLink href='/[slug]' as='/first'>` and `<ActiveLink href='/[slug]' as='/first'>`

Reply:
@xhoantran no need to close the PR! We can apply the above change here if you think the change is alright?

Reply:
@ijjk I understand what you mean now. I just create another [demo](https://stackblitz.com/edit/github-w7g78q?file=components/ActiveLink.js) using `rewrite` and it appears to work like you say.

Reply:
I will add your change and rewrite, then update and will let you know when I finish.